lib/k8s/resource_client.rb

```diff
@@ -200,7 +210,30 @@ def list(labelSelector: nil, fieldSelector: nil, namespace: @namespace)
           'fieldSelector' => selector_query(fieldSelector)
         )
       )
-      process_list(list)
+    end
+
+    # @param labelSelector [nil, String, Hash{String => String}]
+    # @param fieldSelector [nil, String, Hash{String => String}]
+    # @param resourceVersion [nil, String]
+    # @param timeout [nil, Integer]
+    # @raise [Excon::Error]
```

Comment:
Actually, maybe this should be rescued and reraised as some sort of `K8s::TimeoutError` so user doesn't have to care/know about excon.